app/connector/google-sheets/src/main/java/io/syndesis/connector/sheets/GoogleSheetsVerifierExtension.java

```diff
@@ -0,0 +1,63 @@
+package io.syndesis.connector.sheets;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.services.sheets.v4.Sheets;
+import org.apache.camel.CamelContext;
+import org.apache.camel.component.extension.verifier.ResultBuilder;
+import org.apache.camel.component.extension.verifier.ResultErrorBuilder;
+import org.apache.camel.component.google.sheets.BatchGoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsClientFactory;
+import org.apache.camel.component.google.sheets.GoogleSheetsConfiguration;
+
+
+/**
+ * FIXME Remove this class when Camel dependency gets updated
+ **/
+public class GoogleSheetsVerifierExtension
+    extends org.apache.camel.component.google.sheets.GoogleSheetsVerifierExtension {
+    public GoogleSheetsVerifierExtension(String defaultScheme) {
+        super(defaultScheme);
+    }
+
+    public GoogleSheetsVerifierExtension(String defaultScheme, CamelContext context) {
+        super(defaultScheme, context);
+    }
+
+    @Override
+    protected Result verifyConnectivity(Map<String, Object> parameters) {
+        ResultBuilder builder = ResultBuilder.withStatusAndScope(Result.Status.OK, Scope.CONNECTIVITY);
+
+        try {
```

Comment:
We can avoid the nested try-catch when adding a private helper method that constructs the error result from generic exception. Both catch clauses can then call this private helper. Something like:

```
try {
...
} catch (GoogleJsonResponseException e) {
    ...
    if (code != 404) {
        builder.error(getErrorFromException(e));    
    }
} catch (Exception ex) {
    builder.error(getErrorFromException(e));    
}
```

Reply:
Sounds better